service/src/test/java/bio/terra/workspace/pact/landingzones/LzReadApiTest.java

```diff
@@ -0,0 +1,321 @@
+    when(config.getBasePath()).thenReturn(mockServer.getUrl());
+    landingZoneService = new HttpLandingZoneService(OpenTelemetry.noop(), config);
+  }
+
+  @Pact(consumer = "workspacemanager", provider = "terra-landing-zone-service")
+  public RequestResponsePact getLandingZone(PactDslWithProvider builder) {
+    var getResultResponseShape =
+        new PactDslJsonBody()
+            .uuid("landingZoneId")
+            .uuid("billingProfileId")
+            .stringType("definition")
+            .stringType("version")
+            .stringType("region")
+            .date("createdDate");
+
+    return builder
+        .given("An existing landing zone")
+        .uponReceiving("A request to get a landing zone")
+        .method("GET")
+        .pathFromProviderState(
+            "/api/landingzones/v1/azure/${LANDING_ZONE_ID}",
+            "/api/landingzones/v1/azure/%s".formatted(LANDING_ZONE_ID))
+        .willRespondWith()
+        .body(getResultResponseShape)
+        .status(HttpStatus.OK.value())
+        .toPact();
+  }
+
+  @Pact(consumer = "workspacemanager", provider = "terra-landing-zone-service")
+  public RequestResponsePact getLandingZone_notFound(PactDslWithProvider builder) {
+    return builder
+        .uponReceiving("A request for a non-existent landing zone")
+        .method("GET")
+        .path(String.format("/api/landingzones/v1/azure/%s", LANDING_ZONE_ID))
+        .willRespondWith()
+        .status(404)
+        .toPact();
+  }
+
+  @Pact(consumer = "workspacemanager", provider = "terra-landing-zone-service")
+  public RequestResponsePact listLandingZonesForBillingProfile(PactDslWithProvider builder) {
+    var listResponseShape =
+        new PactDslJsonBody()
+            .array("landingzones")
+            .object()
+            .uuid("landingZoneId")
+            .uuid("billingProfileId")
+            .stringType("definition")
+            .stringType("version")
+            .stringType("region")
+            .date("createdDate");
+
+    return builder
+        .given("A landing zone linked to a billing profile")
+        .uponReceiving("A request to list landing zones for a billing profile")
+        .method("GET")
+        .path("/api/landingzones/v1/azure")
+        .query(String.format("billingProfileId=%s", BILLING_PROFILE_ID))
+        .willRespondWith()
+        .body(listResponseShape)
+        .status(HttpStatus.OK.value())
+        .toPact();
+  }
+
+  @Pact(consumer = "workspacemanager", provider = "terra-landing-zone-service")
+  public RequestResponsePact listLandingZones(PactDslWithProvider builder) {
+    var listResponseShape =
+        new PactDslJsonBody()
+            .eachLike("landingzones")
+            .uuid("landingZoneId")
+            .uuid("billingProfileId")
+            .stringType("definition")
+            .stringType("version")
+            .stringType("region")
+            .date("createdDate");
+
+    return builder
+        .uponReceiving("A request to list all landing zones")
```

Comment:
Do we need some sort of state here?

Reply:
@blakery I don't follow, what kind of state?